pkg/segment/aggregations/generateevents.go

```diff
@@ -86,3 +99,112 @@ func PerformGenTimes(aggs *structs.QueryAggregators) error {
 
 	return nil
 }
+
+func checkCSVFormat(filename string) bool {
+	return strings.HasSuffix(filename, ".csv") || strings.HasSuffix(filename, ".csv.gz")
+}
+
+// Ensure record and column lengths are valid before calling this function.
+func createRecord(columnNames []string, record []string) map[string]interface{} {
+	recordMap := make(map[string]interface{})
+	for i, col := range columnNames {
+		recordMap[col] = record[i]
+	}
+	return recordMap
+}
```

Comment:
nit: this might be better as a generic function in maputils.go. But you can keep it here if you prefer

Reply:
addressed in the latest commit.